middle_end/flambda2/tests/lib/minicheck/type.mli

```diff
@@ -0,0 +1,99 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*           Mark Shinwell and Leo White, Jane Street Europe              *)
+(*                                                                        *)
+(*   Copyright 2015--2022 Jane Street Group LLC                           *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+[@@@ocaml.warning "+a-4-30-40-41-42"]
+
+type 'a t
```

Comment:
I still don't love the name `Type`, but the bundling makes sense

Reply:
Maybe `Model`? (That would especially make sense with the new two-argument version, since if you have a `('a, 'repr) Model.t`, `'repr` is basically a model (that is, an abstraction) of `'a`.)

Reply:
What about `Arbitrary` (after QuickCheck's type class), or `Test_input`?  I don't like the generic names, this feels like a much more specific type.

Reply:
I don't love `Test_input` because a `Test_input.t` seems like it would be the actual input to a test, not a description of the _type_ of the input. I suppose a typeclass-style name like `Arbitrary` or `Checkable` might make sense. (Wait, no, `Checkable` sounds like it describes a proposition that can be checked.)

Reply:
Went with `Arbitrary`.